e2etests/test/pnp_connect.js

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+const DeviceIdentityHelper = require('./device_identity_helper.js');
+const createDeviceClient = require('./testUtils.js').createDeviceClient;
+const assert = require('chai').assert;
+
+const deviceMqtt = require('azure-iot-device-mqtt');
+const IoTHubTokenCredentials = require('azure-iothub').IoTHubTokenCredentials;
+const DigitalTwinClient = require('azure-iothub').DigitalTwinClient;
+
+const hubConnectionString = process.env.IOTHUB_CONNECTION_STRING;
+const modelId = 'dtmi:com:example:Thermostat;1';
+
+[
+  DeviceIdentityHelper.createDeviceWithSas,
+  DeviceIdentityHelper.createDeviceWithSymmetricKey,
+  DeviceIdentityHelper.createDeviceWithX509SelfSignedCert,
+  DeviceIdentityHelper.createDeviceWithX509CASignedCert
+].forEach((createDeviceMethod) => {
+  [deviceMqtt.Mqtt, deviceMqtt.MqttWs].forEach((deviceTransport) => {
+    pnpConnectTests(deviceTransport, createDeviceMethod);
+  });
+});
```

Comment:
A suggestion (take it or leave it).  I see this same block of code repeated 5 times.  I feel like it could be a helper function.  (or you could share the 2 arrays that make up the matrix between tests.)

Reply:
Will probably leave it as-is. IMO it's simple enough to where a helper function wouldn't be that useful and might make things a bit less clear

Reply:
Never mind, I made a helper function now that `ModuleClients` are in the mix